doc/source/examples/incompressible-flow/3d-flow-over-periodic-hills/3d-flow-over-periodic-hills.rst

```diff
@@ -0,0 +1,280 @@
+
+* ``repetitions x, repetitions y, repetitions z``: To get cells with an aspect ratio different on the domain, different numbers of subdivisions are set, given by repetition in different coordinate directions. The minimum number of subdivisions in each direction is 1. 
+
+The following image displays a coarse mesh for this example. It can be seen that the horizontal lines are shifted with the associated parameter to get more lines near walls. Here, repetitions for x, y, z allow the mesh to get more cells in x and y directions.
+
+.. image:: images/mesh.png
+    :alt: Mesh
+    :align: center
+    :name: mesh
+
+
+Flow control
+~~~~~~~~~~~~
+
+Since the flow is periodic and a specific Reynolds number is targeted for the simulation, the flow has to be controlled at each time step. To allow flow control, the subsection flow control has to be enabled. 
+
+.. code-block:: text
+
+ #---------------------------------------------------
+ # Flow control
+ #---------------------------------------------------
+ subsection flow control
+     set enable                    = true
+     set boundary id    		   = 0
+     set volumetric flow rate      = -9.1575 # bulk velocity = -1
+     set flow direction 		   = 0
+     set initial beta		       = 7.66
+     set verbosity                 = verbose
+ end
+
+First we set the ``enable`` parameter to ``true`` in order to control the flow. The boundary id ``0`` corresponds to the flow inlet where we want to control the flow. The ``volumetric flow rate`` has to be negative if the flow goes in x positive direction or ``flow direction = 0``. Therefore we adjust this parameter so that we obtain a bulk velocity :math:`u_B` equals to 1. The ``initial beta`` parameter is a coefficient calculated at each time step that speeds up the convergence of the flow rate targeted.
+
+.. tip:: A good method to find a reasonable initial beta is to test two or three different initial beta parameters, write down the given flow rate at the first time step in the simulation and do a regression. The correlation is linear and giving a proper value will greatly speed up the convergence. 
+
+Post-processing
+~~~~~~~~~~~~~~~~~~~
+
+The post-processing subsection allows the calculation of different quantities:
+
+.. code-block:: text
+
+ #---------------------------------------------------
+ # Post-Processing
+ #---------------------------------------------------
+ subsection post-processing
+     set calculate average velocities    = true
+     set initial time 		             = 207
+ end
+
+In this example, we enable the calculation of average velocities through the parameter ``calculate average velocities`` after a certain time of the simulation. In this case, this time is set to ``207`` as we allow for the flow to achieve some stability. The results of the calculated values will be available in the ``.vtu`` files when opened with a visualization software. 
+
+Boundary conditions
```

Comment:
This is just a suggestion, maybe it would be easier to follow it the Boundary condition section was before the Dynamic flow control section, just to keep track of wherer the boundary conditions are with the image.

Reply:
This makes a lot of sense, I will move the section in the next commit